README.md

```diff
@@ -21,238 +24,131 @@ repositories {
 ```
-@Nonnull
-private final ChangesDetector<Data, Data> changesDetector =
-        new ChangesDetector<>(new SimpleDetector<Data>());
+- `itemId` is required to identify specific element in the list. In this case it is the song id as it is unique to the song. This is being used in the `DiffUtil.ItemCallback.areItemsTheSame()` method.
+- `SongItem` is a Kotlin data class so it overrides `equals` and `hashCode` by default. This is being used in the `DiffUtil.ItemCallback.areContentsTheSame()` method to identify whether element's content changed and need to be updated. If you'd like to alter this behaviour you can override `equals` and `hashCode` methods on your own.
+
+## Plug and play data models and view holders
+As your models and view holders are not bound to any adapter, you can reuse them in every adapter.
+You don't have to specify the `itemViewType` in each adapter, just pass you view holder to adapter's constructor.
+
+## Cleaner tests
+As your data models are being created in a ViewModel/Presenter, your test logic is very clean and simple
+
+```kotlin
+@Test
+fun `when 2 of 4 registered students are attendees then student items correctly divided into sections`() {
+    every { classDao.registeredStudents } returns Observable.just(
+        listOf(
+            ClassStudent("name1", attended = true),
+            ClassStudent("name3", attended = false),
+            ClassStudent("name2", attended = true),
+            ClassStudent("name4", attended = false)
+        )
+    )
+    viewModel = PastClassStudentsPresenter("fake_id", classDaos)
+
+    viewModel.adapterItems
+        .test()
+        .assertValue(
+            listOf(
+                SectionNameItem("ATTENDED"),
+                PastClassStudentItem("name1", true),
+                PastClassStudentItem("name2", true),
+                SectionNameItem("REGISTERED"),
+                PastClassStudentItem("name3", false),
+                PastClassStudentItem("name4", false)
+            )
+        )
+}
 ```
 
-Give new data to your adapter:
-
-```java
-List<Data> data = Arrays.toList(new Data(1, "Cow"), new Data(2, "Dg"), new Data(3, "Cat"));
-yourAdapter.swapData(data)
-changesDetector.newData(yourAdapter, data, false);
+## RX support
+`universal-adapter-rx` module lets you subscribe directly to adapter like this:
+```kotlin
+viewModel.adapterItems.subscribe(adapter)
 ```
 
-And another data so recycler view will be nice animated:
-
-```java
-List<Data> data = Arrays.toList(Data(2, "Dog"), new Data(3, "Cat"), new Data(4, "Elephant"));
-yourAdapter.swapData(data)
-changesDetector.newData(yourAdapter, data, false);
-```
 
 ### More
 
-For more:
-- look on sample app at [app/](app/) directory.
-- look on [AutoValue: BaseAdapterItem Extension](https://github.com/m-zagorski/auto-value-base-adapter-item)
-
-
-### Frequently asked questions
-
-[FAQ](FAQ.md)
+For more look at the sample app at [app/](app/) directory.
```

Comment:
suggestion: If you can also update FAQ, it would be cool

Reply:
Yeah I'm gonna update this in another PR soon